### DISCUSSION TITLE
Regexp problem

### DISCUSSION THREAD

**Discussion by bbahes** [variar/klogg#316] [Q&A] [answered]
Hi,

I'm trying to create highlighter for strings that appear in my logs.
I have for example entries like these in logs:
#00000003019e9e0000-0-1
#00000003019e9e0100-0-0
I tried with this ^[#0-9A-Za-z]+[-]+[0-9]+[-]+[0-9]$ but it does not match in klogg.

What am I doing wrong? Wrong regexp?

Thanks!

**Comment by variar** [comment, marked as answer]
Hi, 
What version are you using? Could it be that you have added highlight rule to inactive highlighter set? I've tried your example and it looks ok.


https://user-images.githubusercontent.com/1620716/118862553-8b0ff600-b8e6-11eb-9107-08a98d8fee69.mp4

**Reply by bbahes** [reply]
Highlighter, active:
![image](https://user-images.githubusercontent.com/19546260/118864028-dbd02080-b8df-11eb-9035-4a1883475b86.png)

Version:
![image](https://user-images.githubusercontent.com/19546260/118864140-f99d8580-b8df-11eb-9a9c-9d81b957e05c.png)

**Reply by variar** [reply]
Is this set marked as selected in context menu?

**Reply by bbahes** [reply]
Yes:

![image](https://user-images.githubusercontent.com/19546260/118870400-d0342800-b8e6-11eb-862d-308718da7616.png)

**Reply by variar** [reply]
Try removing `^` and `$` from regex. On the screenshot matching substring is in the middle of text line.

**Reply by bbahes** [reply]
Ha good point...now it work...^is for beginning of the line...
Many thanks for help!
